waiter/src/waiter/process_request.clj

```diff
@@ -175,18 +180,34 @@
                             (log/error e "error releasing instance!"))))
       instance)))
 
+(defn- classify-error
+  "Classifies the error responses from the backend into the following vector:
+   - error cause (:client-error, :instance-error or :generic-error),
+   - associated error message, and
+   - the http status code."
+  [error]
+  (let [classification (cond (instance? ExceptionInfo error)
+                             (let [[error-cause message status] (classify-error (ex-cause error))
+                                   error-cause (or (-> error ex-data :error-cause) error-cause)]
+                               [error-cause message status])
+                             (instance? IllegalStateException error)
+                             [:generic-error (.getMessage error) 400]
+                             (instance? EofException error)
+                             [:client-error "Connection unexpectedly closed while streaming request" 400]
+                             (instance? TimeoutException error)
+                             [:instance-error (utils/message :backend-request-timed-out) 504]
+                             :else
+                             [:instance-error (utils/message :backend-request-failed) 502])
+        [error-cause _ _] classification]
+    (log/info (-> error .getClass .getCanonicalName) (.getMessage error) "identified as" error-cause)
```

Comment:
Are there overloads of the logging functions (e.g. `error`, `log`) that take a throwable?  Is the serialization/formatting of that argument insufficient?  Does it report the class name and message?

Reply:
Placing the raw exception as an argument to the `log/info|error` methods causes the entire stack trace to be printed in various formats, java native stacktrace format or clojure stacktrace format, depending on the position of the argument.

Reply:
Why don't we want the stacktrace?